arch/risc-v/src/common/supervisor/riscv_exception_common.S

```diff
@@ -0,0 +1,137 @@
+/****************************************************************************
+ * arch/risc-v/src/common/supervisor/riscv_exception_common.S
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <arch/irq.h>
+
+#include "riscv_exception_macros.S"
+
+/****************************************************************************
+ * Public Symbols
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: exception_common
+ *
+ * Description:
+ *   Handles delegated interrupts in S-mode interrupt handler.
+ *
+ ****************************************************************************/
+
+  .section .text
+  .global exception_common
+  .align  8
+
+exception_common:
```

Comment:
what's the real difference make the machine and supervisor mode can't share the same code base? I suppose that the macros in mode.h can unify the difference.

Reply:
Don't be afraid to break something, people around the community will review the code and test with their hardware.

Reply:
Modifying code that works is very dangerous as this repository is not managed by amassing requirements and making them into stories etc. There is no clear shared vision, no planning, no product owner, no agile master, no dedicated architect, etc. So there is no one holding the strings, which means a simple change that modifies existing code, can either:

1. Directly break something that works
2. Break some new code under development that is not yet even visible in the repository

Even with a clear organization for planning that can happen, but the risk for this kind of open repository is even greater. This is why I was very hesitant to modify existing code.

Reply:
Done

Reply:
> Modifying code that works is very dangerous as this repository is not managed by amassing requirements and making them into stories etc. There is no clear shared vision, no planning, no product owner, no agile master, no dedicated architect, etc. So there is no one holding the strings, which means a simple change that modifies existing code, can either:
> 
> 1. Directly break something that works
> 2. Break some new code under development that is not yet even visible in the repository
> 
> Even with a clear organization for planning that can happen, but the risk for this kind of open repository is even greater. This is why I was very hesitant to modify existing code.

It depend on how big in the change since it's only a bunch of register remapping(e.g. MSTATUS<->STATUS), the risk is manageable.

Reply:
This is OK. I'm just trying to be careful.